### PR TITLE
ddl: migrate test-infra to testify for db_partition_test

### LINKED ISSUE #28635: migrate test-infra to testify for `ddl/db_partition_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #28635

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

File: testkit/result.go [hunk 1]
```diff
@@ -50,6 +50,25 @@ func (res *Result) Check(expected [][]interface{}) {
 	res.require.Equal(needBuff.String(), resBuff.String(), res.comment)
 }
 
+// CheckAt asserts the result of selected columns equals the expected results.
```
Comment (tisonkun) [inline testkit/result.go:53]: Revert this lines moving only change?

File: testkit/mockstore.go [hunk 2]
```diff
@@ -59,6 +59,29 @@ func bootstrap(t testing.TB, store kv.Storage) (*domain.Domain, func()) {
...
 
+// CreateMockStoreAndDomainWithoutZeroSchemaLease return a new mock kv.Storage and *domain.Domain.
+func CreateMockStoreAndDomainWithoutZeroSchemaLease(t testing.TB, opts ...mockstore.MockTiKVStoreOption) (kv.Storage, *domain.Domain, func()) {
+	store, err := mockstore.NewMockStore(opts...)
+	require.NoError(t, err)
+	dom, clean := bootstrapWithoutZeroSchemaLease(t, store)
+	return store, dom, clean
+}
+
+func bootstrapWithoutZeroSchemaLease(t testing.TB, store kv.Storage) (*domain.Domain, func()) {
```
Comment (tisonkun) [inline testkit/mockstore.go:70]: I think you can pass a variable to set the lease instead and keep the default value to 0.

**Pull request merged**